src/main/java/uk/gov/pay/products/model/Product.java

```diff
@@ -79,6 +80,12 @@ public static Product from(JsonNode jsonPayload) {
                 price, ProductStatus.ACTIVE, gatewayAccountId, serviceName, type, returnUrl);
     }
 
+    public static Product valueOf(ProductEntity productEntity) {
```

Comment:
👎  you should do this in the ProductEntity. Not in Product. The reason being, Product should never need to know about Entity classes and should avoid any dependencies being loaded. 

On the other hand its Okay for ProductEntity to know about Product (so the dependency)

Reply:
It is no longer needed as GatewayAccountUpdater refactored to return a Boolean instead of a List of Products